packages/backend/domain/middleware/auth-middleware.ts

```diff
@@ -1,40 +1,64 @@
 import { CLIENT_ERRORS } from '../errors/client-errors';
 import { Response, NextFunction } from 'express';
 import { QUERY_WITHOUT_AUTH } from '../../api/query-keys/query-keys';
-import jwt from 'jsonwebtoken';
-import { secret } from '../../secret';
 import { CustomRequest } from '../types/custom-request.type';
-import { IRefreshPayload } from '../../api/executors/refresh-token/refresh-token.types';
+import { User } from '../../bd/schemas/user.schema';
+import { IUserBD } from '../../bd/types/user-bd.interface';
+import { SERVER_ERRORS } from '../errors/server-errors';
+import { verifyRefreshToken } from '../../api/executors/refresh-token/verify-refresh-token';
 
-export const authMiddleware = <T>(req: CustomRequest<T>, resp: Response, next: NextFunction): void => {
+export const authMiddleware = async <T>(
+    req: CustomRequest<T>,
+    resp: Response,
+    next: NextFunction
+): Promise<void> => {
     if (req.method === 'OPTIONS') {
         next();
         return;
     }
 
-    const isQueryWithoutAuth = QUERY_WITHOUT_AUTH.includes(req.path);
+    const isQueryWithoutAuth = QUERY_WITHOUT_AUTH.includes(req?.path ?? '');
 
     if (isQueryWithoutAuth) {
         next();
         return;
     }
 
```

Comment:
```suggestion
    if (req.method === 'OPTIONS' || QUERY_WITHOUT_AUTH.includes(req?.path ?? '')) {
        return next();
    }


```suggestion
    if (req.method === 'OPTIONS') {
        next();
        return;
    }

    const isQueryWithoutAuth = QUERY_WITHOUT_AUTH.includes(req?.path ?? '');

    if (isQueryWithoutAuth) {
        next();
        return;
    }

``````